rustls/src/crypto/aws_lc_rs/ticketer.rs

```diff
@@ -51,97 +56,124 @@ impl Ticketer {
 }
 
 fn make_ticket_generator() -> Result<Box<dyn ProducesTickets>, GetRandomFailed> {
-    let mut key = [0u8; 32];
-    SystemRandom::new()
-        .fill(&mut key)
-        .map_err(|_| GetRandomFailed)?;
-
-    let key = aead::UnboundKey::new(TICKETER_AEAD, &key).unwrap();
-
-    let mut key_name = [0u8; 16];
-    SystemRandom::new()
-        .fill(&mut key_name)
-        .map_err(|_| GetRandomFailed)?;
-
-    Ok(Box::new(AeadTicketer {
-        alg: TICKETER_AEAD,
-        key: aead::LessSafeKey::new(key),
-        key_name,
-        lifetime: 60 * 60 * 12,
-        maximum_ciphertext_len: AtomicUsize::new(0),
-    }))
+    // NOTE(XXX): Unconditionally mapping errors to `GetRandomFailed` here is slightly
+    //   misleading in some cases (e.g. failure to construct a padded block cipher encrypting key).
+    //   However, we can't change the return type expected from a `TicketSwitcher` `generator`
+    //   without breaking semver.
+    Ok(Box::new(Rfc5077::new().map_err(|_| GetRandomFailed)?))
 }
 
-/// This is a `ProducesTickets` implementation which uses
-/// any *ring* `aead::Algorithm` to encrypt and authentication
-/// the ticket payload.  It does not enforce any lifetime
-/// constraint.
-struct AeadTicketer {
-    alg: &'static aead::Algorithm,
-    key: aead::LessSafeKey,
+/// An RFC 5077 "Recommended Ticket Construction" implementation of a [`Ticketer`].
+struct Rfc5077 {
+    aes_encrypt_key: cipher::PaddedBlockEncryptingKey,
```

Comment:
What's the value of keeping this type, and a bunch of the other stuff used in `new()`, prefixed with `cipher`?

(For example, I can see the value for using `hmac::Key` but these other identifiers seem pretty specific/unambiguous.)

Reply:
Dropped the `cipher::` prefixes.